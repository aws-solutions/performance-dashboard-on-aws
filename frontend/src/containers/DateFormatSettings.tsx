/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { useHistory } from "react-router-dom";
import { useSettings } from "../hooks";
import dayjs from "dayjs";
import SettingsLayout from "../layouts/Settings";
import Button from "../components/Button";
import Spinner from "../components/Spinner";
import "./PublishedSiteSettings.css";
import { useTranslation } from "react-i18next";

function DateFormatSettings() {
    const history = useHistory();
    const { settings, loadingSettings } = useSettings();
    const { t } = useTranslation();

    const onEdit = () => {
        history.push("/admin/settings/dateformat/edit");
    };

    return (
        <SettingsLayout>
            <h1>{t("SettingsDateTimeFormat")}</h1>

            <p>{t("SettingsDateTimeFormatDescription")}</p>

            {loadingSettings ? (
                <Spinner
                    style={{
                        position: "fixed",
                        top: "50%",
                        left: "50%",
                    }}
                    label={t("LoadingSpinnerLabel")}
                />
            ) : (
                <>
                    <div className="grid-row margin-top-0-important">
                        <div className="grid-col flex-9">
                            <h2>{t("SettingsDateFormat")}</h2>
                        </div>
                        <div className="grid-col flex-3 text-right">
                            <Button className="margin-top-2" variant="outline" onClick={onEdit}>
                                {t("Edit")}
                            </Button>
                        </div>
                    </div>

                    <div className="grid-row margin-top-0-important">
                        <div className="grid-col flex-9">
                            <div className="published-site font-sans-lg">
                                {dayjs()
                                    .locale(window.navigator.language.toLowerCase())
                                    .format(settings.dateTimeFormat.date)}{" "}
                                ({settings.dateTimeFormat.date})
                            </div>
                            <div className="grid-col flex-9">
                                <h2>{t("SettingsTimeFormat")}</h2>
                            </div>
                            <div className="font-sans-lg">
                                {dayjs()
                                    .locale(window.navigator.language.toLowerCase())
                                    .format(settings.dateTimeFormat.time)}{" "}
                                ({settings.dateTimeFormat.time})
                            </div>
                        </div>
                        <div className="grid-col flex-3 text-right"></div>
                    </div>
                </>
            )}
        </SettingsLayout>
    );
}

export default DateFormatSettings;
