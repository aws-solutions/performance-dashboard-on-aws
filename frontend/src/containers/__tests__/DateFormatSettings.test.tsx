/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import DateFormatSettings from "../DateFormatSettings";

jest.mock("../../hooks");
jest.mock("dayjs", () => () => {
    const dayjs = jest.requireActual("dayjs");
    return dayjs("2020-12-09 03:30:00");
});

beforeEach(() => {
    render(<DateFormatSettings />, {
        wrapper: MemoryRouter,
    });
});

test("renders a title", async () => {
    const title = screen.getByRole("heading", { name: "Date and time format" });
    expect(title).toBeInTheDocument();
});

test("renders the date format", async () => {
    const dateFormat = screen.getByText("2020-12-09 (YYYY-MM-DD)");
    expect(dateFormat).toBeInTheDocument();
});

test("renders the time format", async () => {
    const timeFormat = screen.getByText("03:30 (HH:mm)");
    expect(timeFormat).toBeInTheDocument();
});

test("date format settings should match snapshot", async () => {
    const wrapper = render(<DateFormatSettings />, {
        wrapper: MemoryRouter,
    });
    expect(wrapper.container).toMatchSnapshot();
});
