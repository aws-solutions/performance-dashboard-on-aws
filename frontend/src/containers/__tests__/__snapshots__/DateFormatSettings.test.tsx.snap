// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date format settings should match snapshot 1`] = `
<div>
  <div
    class="usa-section padding-top-3"
  >
    <div
      class="grid-container padding-x-0"
    >
      <div
        class="grid-row grid-gap"
      >
        <div
          class="desktop:grid-col-3"
        >
          <nav
            aria-label="Left navigation"
          >
            <ul
              class="usa-sidenav"
            >
              <li
                class="usa-sidenav__item"
              >
                <a
                  class="usa-current"
                  href="/admin/settings"
                >
                  Settings
                </a>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      aria-current="page"
                      class="usa-current"
                      href="/admin/settings/topicarea"
                    >
                      Topic areas
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishingguidance"
                    >
                      Publishing guidance
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/brandingandstyling"
                    >
                      Branding and styling
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishedsite"
                    >
                      Published site
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/dateformat"
                    >
                      Date and time format
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/adminsite"
                    >
                      Admin site
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="desktop:grid-col-9 usa-prose mobile-margin-top-2"
          id="main-content"
        >
          <h1>
            Date and time format
          </h1>
          <p>
            Customize how your performance dashboard displays date and time in the user interface.
          </p>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h2>
                Date format
              </h2>
            </div>
            <div
              class="grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="published-site font-sans-lg"
              >
                2020-12-09
                 
                (
                YYYY-MM-DD
                )
              </div>
              <div
                class="grid-col flex-9"
              >
                <h2>
                  Time format
                </h2>
              </div>
              <div
                class="font-sans-lg"
              >
                03:30
                 
                (
                HH:mm
                )
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
